---
title: "Exam 4"
author: "Ranae Zauner"
date: "April 27, 2018"
output: html_document
---
***
## **Part 1**
#### **Task 1:** Convert these fastq reads to fasta reads. These new fasta reads should be named "Sample1.fasta" and "Sample2.fasta"
##### In Bash:
(Created this .Rmd on a Windows machine, hence this being shown in plain text)

> $ cd c:/Users/Ranae/Desktop/Data_Course/Exams/Exam_3

> $ alias fqtofa='sed -n '\''1~4s/^@/>/p;2~4p'\'''

> $ for fn in *.fastq; do fqtofa $fn > $fn.fasta;done


#### **Task 2:** How many DNA reads were detected in each sample?
##### In Bash:

> $ grep -c "^>" *.fasta

> Returned: ## Sample1.fastq.fasta:6577

> Returned: ## Sample2.fastq.fasta:7731

#### **Task 3:** Trim the original fastq files so that only the first 100 bases are included from each DNA read within.
```{r message=FALSE, warning=FALSE}

library(ShortRead)
library(tidyr)
library(stringr)
library(msa)
library(seqinr)
library(dada2)

setwd("c:/Users/Ranae/Desktop/Data_Course/Exams/Exam_3/")
fq.files = dir(path = getwd(), full.names = TRUE, pattern = ".fastq")

filt.files = dir(path = getwd(), full.names = FALSE, pattern = ".fastq")

dir.create(file.path(getwd(),"filtered"))

# for(i in filt.files){
#   fastqFilter(fn = i,fout = paste0(getwd(),"/filtered/",i,".filt"),
#               truncLen = 100)
# }

```
For some reason, I could not get the for loop to work. It worked in my first attempt at Exam 3, but I can't figure out what the issue is now. I reran everything in bash, had the files, the code prior to the four loop works. No idea. 

##### **Task 4:** Save these trimmed fastq files as "Sample1.fastq.trim" and "Sample2.fastq.trim"  
```{r}
setwd(file.path(paste0(getwd(),"/filtered")))
filtered = dir(path = getwd(), pattern = ".filt")
```
***
## **PART 2:**

Let's get this out of the way:
```{r message=FALSE, warning=FALSE}
library(vegan)
```
##### **Task 1:** Import these files into R objects
```{r}
otu = read.csv("c:/Users/Ranae/Desktop/Data_Course/Exams/Exam_3/exam3_otu_table.csv", row.names = 1)
meta = read.csv("c:/Users/Ranae/Desktop/Data_Course/Exams/Exam_3/exam3_metadata.csv", row.names = 1)
```

##### **Task 2:** Subset both objects to include only samples where SampleType is either "Soil" or "rhizosphere"
```{r}
metaSR = subset(meta, meta$SampleType %in% c("Soil","rhizosphere"))

SR_samples = row.names(metaSR)
otuSR = otu[,names(otu) %in% SR_samples] 
```

##### **Task 3:** Perform a PermANOVA to determine whether there are significant differences in bacterial community composistion between these two sample types

```{r}
t_otuSR = t(otuSR)
adonis(t_otuSR ~ metaSR$SampleType * metaSR$Location)

```
While the SampleType is not important, Location is. The interaction between SampleType and Location is also not important. 

##### **Task 4:** Determine which bacterial genus is the most abundant for each of those sample types as a whole.
The taxonomy was dropped in the otuSR table, so it will have to be pulled from the first table. 

```{r}
metaS = subset(metaSR, SampleType == "Soil")
metaR = subset(metaSR, SampleType == "rhizosphere")

otuS = otuSR[,names(otuSR) %in% row.names(metaS)]
otuR = otuSR[,names(otuSR) %in% row.names(metaR)]
```

```{r}
max_soil = max(rowSums(otuS))
max_rhiz = max(rowSums(otuR))

max_soil_taxonomy = which(rowSums(otuS) == max_soil)
max_rhiz_taxonomy = which(rowSums(otuR) == max_rhiz)

# Use this to pull out taxonomic assignments from original OTU table
otu$Consensus.lineage[max_soil_taxonomy]
otu$Consensus.lineage[max_rhiz_taxonomy]
```









