

##EXAM#: PART 1
# <Bash>
# >>>>Task 1: Convert these fastq reads to fasta reads. These new fasta reads should be named "Sample1.fasta" and 
# "Sample2.fasta"
# $ cd c:/Users/Ranae/Desktop/Data_Course/Exam3
# $ alias fqtofa='sed -n '\''1~4s/^@/>/p;2~4p'\'''
# $ for fn in *.fastq; do fqtofa $fn > $fn.fasta;done


# >>>> Task 2: How many DNA reads were detected in each sample?
# $ grep -c "^>" *.fasta
## Sample1.fastq.fasta:6577
## Sample2.fastq.fasta:7731

# Task 3: Trim the original fastq files so that only the first 100 bases are included from each DNA read within.

```{r eval=FALSE, include=FALSE}
library(ShortRead)
library(tidyr)
library(stringr)
library(msa)
library(seqinr)
library(dada2)
setwd("c:/Users/Ranae/Desktop/Data_Course/Exam3/")
fq.files = dir(path = getwd(), full.names = TRUE, pattern = ".fastq")

filt.files = dir(path = getwd(), full.names = FALSE, pattern = ".fastq")

dir.create(file.path(getwd(),"filtered"))

for(i in filt.files){
  fastqFilter(fn = i,fout = paste0(getwd(),"/filtered/",i,".filt"),
              truncLen = 100)
}
```  
  
# Task 4: Save these trimmed fastq files as "Sample1.fastq.trim" and "Sample2.fastq.trim"  

```{r eval=FALSE, include=FALSE}

setwd(file.path(paste0(getwd(),"/filtered")))
filtered = dir(path = getwd(), pattern = ".filt")
  
```

#EXAM 3: PART 2

# Task 1: Import these files into R objects
```{r}
library(vegan)
otu_table = read.csv("c:/Users/Ranae/Desktop/Data_Course/Exam3/exam3_otu_table.csv", row.names = 1)
metadata = read.csv("c:/Users/Ranae/Desktop/Data_Course/Exam3/exam3_metadata.csv", row.names = 1)
```

# Task 2: Subset both objects to include only samples where SampleType is either "Soil" or "rhizosphere"

```{r}
t_otu = as.data.frame(t(otu_table))
sub_meta <- subset(metadata, SampleType ==c("Soil","rhizosphere"))
sub_otu <- subset(t_otu, metadata$SampleType==c("Soil","rhizosphere"))
```

# Task 3: Perform a PermANOVA to determine whether there are significant differences in bacterial community composistion between these two sample types

```{r eval=FALSE, include=FALSE}
identical(row.names(sub_otu), as.character(sub_meta$SampleType))

tsub_otu <- t(sub_otu)
msub_otu <- as.numeric(sub_otu)

adonis(msub_otu ~ sub_meta$SampleType)
#Error in rowSums(x, na.rm = TRUE) : 'x' must be numeric
#I have tried transposing my sub_otu, as.numeric, as.matrix... I can't seem to get this to work. 

```
# Task 4: Determine which bacterial genus is the most abundant for each of those sample types as a whole

```{r eval=FALSE, include=FALSE}
diversity(sub_otu)
#Error, input data must be non-negative
#Again, no idea why I can't this to work either. 
```
